src/audio/src/src_uuid.h

```diff
@@ -0,0 +1,32 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Baofeng Tian <baofeng.tian@intel.com>
+ */
+
+#ifndef __SOF_AUDIO_SRC_UUID_H__
+#define __SOF_AUDIO_SRC_UUID_H__
+
+#include <sof/lib/uuid.h>
+#include <sof/trace/trace.h>
+#include <sof/audio/src/src.h>
+
+#if CONFIG_IPC_MAJOR_4
+
+/* e61bb28d-149a-4c1f-b709-46823ef5f5a3 */
+DECLARE_SOF_RT_UUID("src", src_uuid, 0xe61bb28d, 0x149a, 0x4c1f,
```

Comment:
the name is confusing, this isn't a declaration, it's a definition - it actually defines objects. So, this shouldn't be in a header. Besides - this can be perfectly placed in src_ipc4.c and src_ipc3.c, they divide very well.

Reply:
I also want to put it ipc3 and ipc4 separately, however, if you check DECLARE_SOF_RT_UUID inside, you will find static definition, how it can be referred by src.c?

#define DECLARE_SOF_UUID(entity_name, uuid_name,			\
			 va, vb, vc,					\
			 vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7)	\
	__section(".static_uuids")					\
	static const struct sof_uuid_entry uuid_name ## _ldc = {	\
		{.a = va, .b = vb, .c = vc,				\
		 .d = {vd0, vd1, vd2, vd3, vd4, vd5, vd6, vd7}},	\
		entity_name "\0"					\
	}

Reply:
right, that defined an object called `src_uuid_ldc`, right? And that one is then used in `DECLARE_TR_CTX(src_tr, SOF_UUID(src_uuid), LOG_LEVEL_INFO);` So I would either duplicate that single line `DECLARE_TR_CTX(...);` in both ipc3 and ipc4 files or leave them all in src.c with `#if`s but still please don't include them in a header. By placing them in that header and **only** including it in src.c you effectively placing them there, so you can just do that explicitly instead.

Reply:
there is another way to use src_uuid:
DECLARE_MODULE_ADAPTER(src_interface, src_uuid, src_tr);
hard to resolve this one.

Reply:
then have to add extern part, I added, please check again, not sure this looks better or not.

Reply:
Yes, that now looks correct C. Someone might complain that macros like `DECLARE_SOF_RT_UUID()` and `DECLARE_TR_CTX()` are supposed to hide their implementation and you aren't supposed to access internals of the code, that they generate, but I'd say that isn't a major concern.